### PR TITLE
Mig/delete old scenariorun prod 9734

### PR DESCRIPTION
**Pull request by mahkoCosmo**

### CODE REVIEW

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt [hunk 1]
```diff
@@ -383,9 +386,70 @@ internal class ScenarioRunServiceImpl(
...
 
+  private fun deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(
+      currentRun: ScenarioRun,
+      lastRun: ScenarioLastRun?
+  ) {
+    while (true) {
+      val scenarioRunStatus = getScenarioRunStatus(currentRun.organizationId!!, currentRun.id!!)
+
+      if (scenarioRunStatus.state == ScenarioRunState.Successful) {
+        // deleteScenarioRunsByScenario delete all previous probes data, if it's what we want
```
Comment (vcarluer) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt:412]: Yes this is what we want => no need to create and call the new method deleteLastRunSimulationData

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt [hunk 2]
```diff
@@ -383,9 +386,70 @@ internal class ScenarioRunServiceImpl(
...
+        return
+      } else if (scenarioRunStatus.state == ScenarioRunState.DataIngestionFailure ||
+          scenarioRunStatus.state == ScenarioRunState.Failed) {
+        logger.info("ScenarioRun {} failed", currentRun.csmSimulationRun)
+        break
+      }
+    }
+  }
+
+  private fun deleteLastRunSimulationData(currentRun: ScenarioRun, lastRun: ScenarioLastRun?) {
```
Comment (vcarluer) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt:427]: not needed

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt (hunk 1 above)
Comment (vcarluer) [reply scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt:412]: No more need need of lastRun also with deleteScenarioRunsByScenario

**Review by vcarluer — CHANGES_REQUESTED** [review]
reuse @LeopoldCramer method

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt [hunk 3]
```diff
@@ -383,9 +386,70 @@ internal class ScenarioRunServiceImpl(
...
+          deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(
+              scenarioRun, startInfo.scenario.lastRun)
+        }
+      }
+      logger.debug("Coroutine to poll simulation status launched")
+    }
     return scenarioRun.withoutSensitiveData()!!
   }
 
+  private fun deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(
```
Comment (jreynard-code) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt]: Please no `while(true)` it's awful.
Try something like this:
```
  private fun deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(currentRun: ScenarioRun) {
    val scenarioRunId = currentRun.id!!
    val workspaceId = currentRun.workspaceId!!
    val organizationId = currentRun.organizationId!!
    val scenarioId = currentRun.scenarioId!!
    val csmSimulationRun = currentRun.csmSimulationRun

    var scenarioRunStatus = getScenarioRunStatus(organizationId, scenarioRunId)
    while (
      scenarioRunStatus.state != ScenarioRunState.Successful ||
      scenarioRunStatus.state != ScenarioRunState.Failed ||
      scenarioRunStatus.state != ScenarioRunState.DataIngestionFailure ) {
      logger.info("ScenarioRun {} is still running, waiting for purging data", csmSimulationRun)
      Thread.sleep(10000)
      scenarioRunStatus = getScenarioRunStatus(organizationId, scenarioRunId)
    }

    if (scenarioRunStatus.state == ScenarioRunState.Successful) {
      logger.info("ScenarioRun {} is Successfull => purging data", csmSimulationRun)
      deleteScenarioRunsByScenario(organizationId, workspaceId, scenarioId)
    } else {
      logger.info("ScenarioRun {} is in error {} => no purging data",
        csmSimulationRun,
        scenarioRunStatus.state)
    }

  }
```
Note also that:  Thread.sleep(1000) should be configurable

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt [hunk 4]
```diff
@@ -383,9 +385,36 @@ internal class ScenarioRunServiceImpl(
...
+        withTimeout(CSM_ARGO_WORKFLOWS_TIMEOUT.toLong()) {
+          deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(scenarioRun)
+        }
+      }
+      logger.debug("Coroutine to poll simulation status launched")
+    }
     return scenarioRun.withoutSensitiveData()!!
   }
 
+  private fun deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(currentRun: ScenarioRun) {
```
Comment (jreynard-code) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt:402]: Please do not write `while(true)` it's awful.
try something like this
```
  private fun deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(currentRun: ScenarioRun) {
    val scenarioRunId = currentRun.id!!
    val workspaceId = currentRun.workspaceId!!
    val organizationId = currentRun.organizationId!!
    val scenarioId = currentRun.scenarioId!!
    val csmSimulationRun = currentRun.csmSimulationRun

    var scenarioRunStatus = getScenarioRunStatus(organizationId, scenarioRunId)
    while (
      scenarioRunStatus.state != ScenarioRunState.Successful ||
      scenarioRunStatus.state != ScenarioRunState.Failed ||
      scenarioRunStatus.state != ScenarioRunState.DataIngestionFailure ) {
      logger.info("ScenarioRun {} is still running, waiting for purging data", csmSimulationRun)
      Thread.sleep(10000)
      scenarioRunStatus = getScenarioRunStatus(organizationId, scenarioRunId)
    }

    if (scenarioRunStatus.state == ScenarioRunState.Successful) {
      logger.info("ScenarioRun {} is Successfull => purging data", csmSimulationRun)
      deleteScenarioRunsByScenario(organizationId, workspaceId, scenarioId)
    } else {
      logger.info("ScenarioRun {} is in error {} => no purging data",
        csmSimulationRun,
        scenarioRunStatus.state)
    }

  }
  ```
  Note that the value 10000 should be configurable

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt [hunk 5]
```diff
@@ -64,7 +66,7 @@ private const val MIN_SDK_VERSION_MINOR = 5
 
 @Service
 @ConditionalOnProperty(name = ["csm.platform.vendor"], havingValue = "azure", matchIfMissing = true)
-@Suppress("TooManyFunctions")
+@Suppress("TooManyFunctions", "LargeClass")
```
Comment (jreynard-code) [inline scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt:69]: Check if after review, the "LargeClass" is still relevant

**Review by jreynard-code — CHANGES_REQUESTED** [review]
Some changes ;)

File: scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt (hunk 4 above)
Comment (mahkoCosmo) [reply scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt:402]: It's a sore in the eye
with the following code the intention is clear and neat

 private fun deletePreviousSimulationDataIfCurrentSimulationIsSuccessful(currentRun: ScenarioRun) {
    while (true) {
      Thread.sleep(10000)
      val scenarioRunStatus = getScenarioRunStatus(currentRun.organizationId!!, currentRun.id!!)

      if (scenarioRunStatus.state == ScenarioRunState.Successful) {
        deleteScenarioRunsByScenario(
            currentRun.organizationId!!, currentRun.workspaceId!!, currentRun.scenarioId!!)
        return
      } else if (scenarioRunStatus.state == ScenarioRunState.DataIngestionFailure ||
          scenarioRunStatus.state == ScenarioRunState.Failed) {
        logger.info("ScenarioRun {} failed", currentRun.csmSimulationRun)
        break
      }
    }
  }

**Review by vcarluer — APPROVED** [review]

**Review by jreynard-code — APPROVED** [review]
LGTM

**Pull request merged**